test/functional/rpc_getbalances.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the getbalances RPC.
+
+- Generate a few blocks
+- Check for certain fields in returned JSON
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (assert_equal, assert_greater_than,
+                                assert_is_hash_string)
+
+class GetBalancesTest (BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        # Move the chain
+        self.nodes[0].generatetoaddress(2, self.nodes[0].get_deterministic_priv_key().address)
+        # Check JSON outputs
+        balances = self.nodes[0].getbalances()
+        assert_greater_than(balances['mine']['immature'], 0)
+        assert_equal(balances['mine']['trusted'], 0)
+        # Check if lastblock present and of type hash
+        lastblock1 = balances['lastblock']
+        assert_is_hash_string(lastblock1)
```

Comment:
Compare with `self.nodes[0].getbestblockhash()` instead?

Reply:
Ok, will update the tests.